cmd/ferretdb/main.go

```diff
@@ -381,6 +402,19 @@ func run() {
 
 	logger := setupLogger(stateProvider, cli.Log.Format)
 
+	otelShutdown := setupTracer()
+
+	if otelShutdown != nil {
+		defer func() {
```

Comment:
It should be moved below to use context from `ctxutil.SigTerm`

Reply:
Isn't it a different context? When `SigTerm` is executed, we need to give more time to let parts of FerretDB stop gracefully. This is why we use a new context with timeout - and only if shutdown didn't happen within the timeout, shutdown is interrupted. Using `SigTerm`'s context will stop it immediately, without a graceful approach.

Reply:
Then I guess we need to change "setup"/"shutdown" with "run until ctx is canceled" like other things there

Reply:
but who will decide when is ctx canceled? :)

Reply:
The user will decide – by sending SIGTERM or another signal. Or I don't understand the question.

Note how context from `ctxutil.SigTerm` is passed to `debug.RunHandler`, `runTelemetryReporter`, etc. We should do the same.